nobl9/resource_agent.go

```diff
@@ -296,25 +301,28 @@ func unmarshalAgent(d *schema.ResourceData, objects []n9api.AnyJSONObj) diag.Dia
 
 func unmarshalAgentConfig(
 	d *schema.ResourceData,
-	object n9api.AnyJSONObj,
+	agent n9api.Agent,
 	hclName,
 	jsonName string) (bool, diag.Diagnostics) {
 	var diags diag.Diagnostics
-	spec := object["spec"].(map[string]interface{})
-	if spec[jsonName] == nil {
-		return false, nil
-	}
 
 	// err := d.Set("agent_type", spec[""]) TODO
-	err := d.Set("source_of", spec["sourceOf"])
+	err := d.Set("source_of", agent.Spec.SourceOf)
+	diags = appendError(diags, err)
+
+	spec, err := json.Marshal(&agent.Spec)
+	diags = appendError(diags, err)
+
+	var m map[string]interface{}
+	err = json.Unmarshal(spec, &m)
```

Comment:
I know it's dirty, but I believe it's the fastest way to achieve this without introducing all the `unmarshalAgentXSpec` methods... Let me know what do you think about it.

Maybe this could be rewritten soon-ish once someone from different domain starts working to fix the lack of `historicalDataRetrieval` support in Agent resource.

Reply:
I think it is fine, I guess. We can do it later.